FreeRTOS/FreeRTOS-Kernel
# Add configUSE_CORE_AFFINITY bits check

Description
-----------
Add `configUSE_CORE_AFFINITY `bits check
* UBaseType_t should have more bits than configNUMBER_OF_CORES
* Single core FreeRTOS should set this macro to 0

Test Steps
-----------
There will be compilation error with the following configurations:
* configNUMBER_OF_CORES = 1
* configUSE_CORE_AFFINITY = 1

```
In file included from C:/msys64/home/chinglee/kernel/init_review/test/FreeRTOS/FreeRTOS/Demo/Common/Minimal/IntSemTest.c:36:
C:/msys64/home/chinglee/kernel/init_review/test/FreeRTOS/FreeRTOS/Source/include/FreeRTOS.h:1169:6: error: #error configUSE_CORE_AFFINITY is not supported in single core FreeRTOS
 1169 |     #error configUSE_CORE_AFFINITY is not supported in single core FreeRTOS
      |      ^~~~~
ninja: build stopped: subcommand failed.
```

Checklist:
----------



Related Issue
-----------
Address #750 comment in line [307](https://github.com/RichardBarry/FreeRTOS-Kernel/blob/814f078563eaf20ba9f59bd36cf48d85ae5e3185/tasks.c#L307C82-L307C245)


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
